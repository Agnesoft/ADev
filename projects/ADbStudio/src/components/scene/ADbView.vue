<template>
    <div class="adb-view" :class="{'search-mode': searchedWord !== ''}">
        <div class="view">
            <Edge v-for="(edges,key) in edgesGrouped" :key="key" :elements="edges" :viewSize="viewSize" :class="{selected: edgeIsSelected(edges)}" v-on="$listeners" />
            <Node v-for="node in nodes" :key="node.id" :element="node" :viewSize="viewSize" :class="{selected: node.id == searchedWord}" v-on="$listeners" />
        </div>
    </div>
</template>

<script>
import Node from "./Node";
import Edge from "./Edge";
import { mapGetters } from 'vuex'

export default {
    name: "ADbView",
    components: {
        Node,
        Edge,
    },
    data(){
        return {
            viewWidth: 0,
            viewHeight: 0,
        }
    },
    props: {
        searchedWord: String,
    },
    computed: {
        ...mapGetters({
            nodes: 'scene/getNodes',
            edges: 'scene/getEdges',
            edgesGrouped: 'scene/getEdgesGrouped',
        }),
        viewSize(){
            return {
                width: this.viewWidth,
                height: this.viewHeight
            }
        }
    },
    methods: {
        onResize() {
            this.viewWidth = document.documentElement.clientWidth;
            this.viewHeight = document.documentElement.clientHeight;
        },
        edgeIsSelected(edges){
            return edges.find((edge) => edge.id == this.searchedWord) ? true : false;
        }
    },
    mounted() {
        window.addEventListener('resize', this.onResize)
        this.onResize();
    },

    beforeDestroy() {
        window.removeEventListener('onmousewheel', this.onResize)
    }

}
</script>

<style scoped>
    .adb-view{
        position: relative;
        transform: scale(var(--scale,1)) translate(var(--move-x,0), var(--move-y,0));
        width: 100%;
        height: 100%;
    }
</style>
