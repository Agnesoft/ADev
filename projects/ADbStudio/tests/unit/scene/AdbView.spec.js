import { mount  } from '@vue/test-utils';
import Vuex from 'vuex';
import ADbView from '@/components/scene/ADbView.vue';
import scene_store from '../../../src/store/modules/scene'

import { localVue,scene_state } from '../TestUtils'


describe('ADbView',() => {
    let wrapper;
    let store;

    beforeAll(() => {
        localVue.use(Vuex)
        
        store = new Vuex.Store({
            modules: {
                scene: {
                    namespaced: true,
                    state: scene_state,
                    getters: scene_store.getters
                }
              }
        });
        wrapper = mount(ADbView, {
            localVue,
            store
          });
    })
    it('renders nodes and edges',() => {
        expect(wrapper.find('.node').exists()).toBe(true);
        expect(wrapper.find('.edge').exists()).toBe(true);
    })
})
