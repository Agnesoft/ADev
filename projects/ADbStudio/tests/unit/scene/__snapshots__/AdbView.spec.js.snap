// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ADbView renders correctly (snapshot) 1`] = `
<div
  class="adb-view search-mode"
>
  <div
    class="view"
  >
    <div
      class="edge"
      style="--top_pos: 0px; --left_pos: 0px; --length: 0px; --angle: 0rad;"
    >
      <!---->
       
      <div
        class="arrow front"
      />
       
      <div
        class="element"
      >
        
    -1

      </div>
    </div>
     
    <div
      class="node"
      style="--node-size: 2.4rem; top: calc(0px - 1.2rem); left: calc(0px - 1.2rem);"
    >
      <div
        class="element"
      >
        
    1

      </div>
    </div>
    <div
      class="node"
      style="--node-size: 2.4rem; top: calc(0px - 1.2rem); left: calc(0px - 1.2rem);"
    >
      <div
        class="element"
      >
        
    2

      </div>
    </div>
  </div>
</div>
`;
